components/reports/reports-filters.tsx

```diff
@@ -1,212 +1,228 @@
-		if (outletsSelected.length) {
-			filtersApplied++;
-			if (searchParams.has("outlet")) {
-				searchParams.delete("outlet");
-			}
-			for (let i = 0; i < outletsSelected.length; i++) {
-				searchParams.append("outlet", outletsSelected[i]);
-			}
-		}
-		if (statesSelected.length) {
-			filtersApplied++;
-			if (searchParams.has("state")) {
-				searchParams.delete("state");
-			}
-			for (let i = 0; i < statesSelected.length; i++) {
-				searchParams.append("state", statesSelected[i]);
-			}
-		}
-		setNumFiltersApplied(filtersApplied);
-		setSearchParams(searchParams);
-    router.push(`reports/?${searchParams.toString()}`, {scroll: false})
-	};
+	const { filters, numFiltersApplied } = useFilters();
 
 	return (
 		<div>
 			<div className="md:hidden">
 				<Drawer>
-					<DrawerTrigger
-						className="flex gap-2 h-10 w-full rounded-md border-input justify-between items-center bg-vd-beige-100 border border-vd-blue-500 px-3 py-2"
-						onClick={() => {
-							setAmountRangeSelected([minAmountNeeded, maxAmountNeeded]);
-							setOutletsSelected([]);
-							setStatesSelected([]);
-						}}
-					>
-						<p className="text-sm font-medium text-vd-blue-500">Filters</p>
-						{numFiltersApplied ? (
-							<div className="bg-vd-blue-100 rounded-full text-xs font-medium text-vd-blue-500 px-2 py-1">
+					<div className="relative">
+						{numFiltersApplied !== 0 && (
+							<div className="bg-vd-blue-100 rounded-full text-xs font-medium text-vd-blue-500 px-2 py-1 h-6 w-6 absolute -right-2 -top-2">
 								{numFiltersApplied}
 							</div>
-						) : (
-							<Filter color="#4B778F" size={16} />
 						)}
-					</DrawerTrigger>
-					<DrawerContent className="">
-						<FilterItems
-							outlets={outlets}
-							states={states}
-							minAmountNeeded={minAmountNeeded}
-							maxAmountNeeded={maxAmountNeeded}
-							amountRangeSelected={amountRangeSelected}
-							setAmountRangeSelected={setAmountRangeSelected}
-							setStatesSelected={setStatesSelected}
-							outletsSelected={outletsSelected}
-							setOutletsSelected={setOutletsSelected}
-						/>
-						<DrawerFooter className="flex-row justify-center gap-2 pb-8">
+						<DrawerTrigger
+							className="flex gap-2 h-10 rounded-md border-input justify-between
+							items-center bg-vd-beige-100 border border-vd-blue-500 px-3 py-2
+							text-vd-blue-500 hover:text-vd-blue-100 overflow-visible"
+						>
+							<Filter size={16} />
+							<p className="text-sm font-medium">Filters</p>
+						</DrawerTrigger>
+					</div>
+					<DrawerContent className="px-6 pb-3">
+						<FilterItems filterOptions={filterOptions} />
+						{/* <DrawerFooter className="flex-row justify-center gap-2 pb-8">
 							<DrawerClose>
 								<Button
 									className="px-24 py-4"
-									onClick={() => handleApplyFilters()}
 								>
 									Apply
 								</Button>
 							</DrawerClose>
-						</DrawerFooter>
+						</DrawerFooter> */}
```

Comment:
removal of "apply" button / separate handling of each filter selection causes mobile drawer to close on any click inside the open drawer. This allows the user to only apply a single filter each time the filter drawer is opened.